### PR TITLE
encryption: Lazily decrypt `webhook_logs`

### PR DESCRIPTION
**Pull request by efritz**
Stacked on #40282. This PR wraps the `webhook_logs.request` and `webhook_logs.response` fields in an encryptable struct.

## Test plan

main-dry-run: https://buildkite.com/sourcegraph/sourcegraph/builds/166837

### CODE REVIEW

File: internal/types/webhook_logs.go [hunk 1]
```diff
@@ -21,3 +23,18 @@ type WebhookLogMessage struct {
...
 	Version string `json:",omitempty"`
 }
+
+type EncryptableWebhookLogMessage = encryption.JSONEncryptable[WebhookLogMessage]
+
+func NewUnencryptedWebhookLogMessage(value WebhookLogMessage) *EncryptableWebhookLogMessage {
+	return NewUnencryptedWebhookLogMessageWithKey(value, nil)
+}
+
+func NewUnencryptedWebhookLogMessageWithKey(value WebhookLogMessage, key encryption.Key) *EncryptableWebhookLogMessage {
```
Comment (vrto) [inline internal/types/webhook_logs.go:33]: Does this need to be exported?

**Review by vrto — APPROVED** [review]
Looking good, just one question!

File: internal/types/webhook_logs.go (hunk 1 above)
Comment (efritz) [reply internal/types/webhook_logs.go:33]: It doesn't, all of the constructors from this and sibling PRs were copy-pasted from the original `Encryptable` draft. Can clean up.

Comment (efritz) [reply internal/types/webhook_logs.go]: Updated. Updating sibling PRs now as well.

**Pull request merged**